### PR TITLE
add steps for CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by zitongzhan**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,8 +1,123 @@
...
+This document covers a step-by-step guide of contributing. It will tell you how to start from setting up the code repository and create a Pull Request step by step. This guide is modified from [OpenMMlab](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md)'s contribution guide.
+
+### 1. Fork and clone
+
+If you are posting a pull request for the first time, you should fork the PyPose repository by clicking the **Fork** button in the top right corner of the GitHub page, and then the forked repositories will appear under your GitHub profile.
+
+Then, you can clone the repositories to local:
+
+```shell
+git@github.com:{your_github_id}/pypose.git
```
Comment (wang-chen) [inline CONTRIBUTING.md:24]: This doesn't work for me. How about `git clone https://github.com/{your_github_id}/pypose.git`.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,8 +1,123 @@
...
+
+### 2. Create a development branch
+After cloning a local copy of the repo, we should create a branch based on the master branch to develop the new feature or fix the bug. The proposed branch name is `username/pr_name`
+
+```shell
+git checkout -b {your_github_id}/{pr_name}
+```
+
+In subsequent development, if the master branch of the local repository is behind the master branch of "upstream", we need to pull the upstream for synchronization, and then execute the above command:
```
Comment (wang-chen) [inline CONTRIBUTING.md:51]: Our default branch is `main` branch, not `master` branch. need to change all `master` to `main` in this readme file.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,8 +1,123 @@
...
+Then, you can clone the repositories to local:
+
+```shell
+git@github.com:{your_github_id}/pypose.git
+```
+
+After that, you should add official repository as the upstream repository
+
+```bash
+git remote add upstream git@github.com:pypose/pypose.git
```
Comment (wang-chen) [inline CONTRIBUTING.md:30]: this doesn't work for me either. how about `git remote add upstream https://github.com/pypose/pypose.git`

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,8 +1,123 @@
...
+```
+
+Check whether remote repository has been added successfully by `git remote -v`
+
+```bash
+origin	git@github.com:{your_github_id}/pypose.git (fetch)
+origin	git@github.com:{your_github_id}/pypose.git (push)
+upstream	git@github.com:pypose/pypose.git (fetch)
+upstream	git@github.com:pypose/pypose.git (push)
```
Comment (wang-chen) [inline CONTRIBUTING.md:39]: Then this becomes
```bash
origin	https://github.com/{your_github_id}/pypose.git (fetch)
origin	https://github.com/{your_github_id}/pypose.git (push)
upstream	https://github.com/pypose/pypose.git (fetch)
upstream	https://github.com/pypose/pypose.git (push)
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,8 +1,123 @@
...
+```
+
+> Here's a brief introduction to origin and upstream. When we use "git clone", we create an "origin" remote by default, which points to the repository cloned from. In the above example, "origin" is your own forked repo. As for "upstream", we add it ourselves to point to the target repository, which is PyPose's official repo `pypose/pypose`. Of course, you could name it to any arbitrary name other than `upstream`. Usually, contributors will push the code to "origin". If the pushed code conflicts with the latest code in official("upstream"), the contributor should pull the latest code from upstream to resolve the conflicts, and then push to "origin" again. The posted Pull Request will be updated automatically.
+
+### 2. Create a development branch
+After cloning a local copy of the repo, we should create a branch based on the master branch to develop the new feature or fix the bug. The proposed branch name is `username/pr_name`
+
+```shell
+git checkout -b {your_github_id}/{pr_name}
```
Comment (wang-chen) [inline CONTRIBUTING.md:48]: The developers don't have to use a development branch within the forked repo. They can use their own repo's `main` branch and create a pull request from that.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,8 +1,123 @@
...
+
+### 5. Create a Pull Request
+
+(1) Create a pull request in GitHub's Pull request interface
+
+(2) Modify the PR description accordingly so that other developers can better understand your changes
+
+**note**
+
+(a) The Pull Request description should contain the reason for the change, the content of the change, and the impact of the change, and be associated with the relevant Issue (see [documentation](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
```
Comment (wang-chen) [inline CONTRIBUTING.md:87]: need another right bracket `)` at the end of this line.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -1,8 +1,123 @@
...
+
+(b) Check whether the Pull Request pass through the CI
+
+TBD
+
+CI will run unit test for the posted Pull Request on different platforms (Linux, Window, Mac), based on different versions of Python, PyTorch, CUDA to make sure the code is correct. We can see the specific test information by clicking `Details` in the above image so that we can modify the code.
```
Comment (wang-chen) [inline CONTRIBUTING.md:93]: remove this if we don't have currently.

or use 
image
by  using this code
```
<img width="909" alt="image" src="https://user-images.githubusercontent.com/8695500/222796987-3f612b21-21c9-4bb4-b8ff-4b1ddccf115d.png">
```

**Review by wang-chen — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 5 above)
Comment (zitongzhan) [reply CONTRIBUTING.md:48]: Is it necessary to keep the step, to prevent people who are in the PyPose organization accidentally push to `pypose/main`?

Comment (wang-chen) [reply CONTRIBUTING.md:48]: Oh, they won't succeed, the current settings need at least 1 reviewer approval to merge a pull request and no one can directly push to `main` branch.
but yes, for people in PyPose Github organization, they should use a `dev` branch.

**Review by wang-chen — APPROVED** [review]

**Pull request merged**